athena-scripts/raw_stage_optimisation_solution/scripts/core_preprocessing_functions.py

```diff
@@ -166,14 +167,43 @@ def generate_raw_select_filter(json_data, database, table, filter_processed_dt):
         if event_processing_selection_criteria_limit is None:
             raise ValueError("filter value for event_processing_testing_criteria is not found within config rules")
         print(f'config rule: event_processing_testing_criteria | filter: {event_processing_testing_criteria_filter}')
-
-        sql=f'''select * from \"{database}\".\"{table}\"'''
-
-        if event_processing_testing_criteria_enabled and event_processing_testing_criteria_filter is not None:
-            sql = sql + f' where {event_processing_testing_criteria_filter}'
+        
+        event_processing_view_criteria_enabled = extract_element_by_name(json_data, "enabled", "event_processing_view_criteria")
+        if event_processing_view_criteria_enabled is None:
+            raise ValueError("enabled value for event_processing_view_criteria is not found within config rules")
+        print(f'config rule: event_processing_view_criteria | enabled: {event_processing_view_criteria_enabled}')
+
+        event_processing_view_criteria_view = extract_element_by_name(json_data, "view_name", "event_processing_view_criteria")
+        if event_processing_view_criteria_view is None:
+            raise ValueError("filter value for event_processing_view_criteria is not found within config rules")
+        print(f'config rule: event_processing_view_criteria | view: {event_processing_view_criteria_view}')
+        
+        
+        deduplicate_subquery = f'''select *,
+			            row_number() over (
+				            partition by event_id
+				            order by cast(
+                                        concat(
+                                            cast(year as varchar),
+                                            cast(lpad(cast(month as varchar), 2, '0') as varchar), 
+                                            cast(lpad(cast(day as varchar), 2, '0') as varchar)
+                                        ) as int
+                                    ) desc
+			                ) as row_num
+               		    from \"{database}\".\"{table}\" as t '''
+                     
+        sql = f'''select * from \"{database}\".\"{table}\"'''
+        
+        if event_processing_view_criteria_enabled and event_processing_view_criteria_view is not None:
```

Comment:
i don't know python well so it may be fine but i get the impression this is a boolean check but `event_processing_view_criteria_enabled` is a string? or is it like js double equals?